spec/facter/resolvers/ec2_spec.rb

```diff
@@ -130,7 +131,17 @@
 
     let(:token) { nil }
     let(:headers) { { 'Accept' => '*/*' } }
+    let(:expected_header) { {} }
 
     it_behaves_like 'ec2'
   end
+
+  it 'does not add headers if token is nil' do
+    allow(Facter::Resolvers::Ec2).to receive(:v2_token).and_return(nil)
+
+    stub_request(:any, metadata_uri).with { |request| !request.headers.key?('X-aws-ec2-metadata-token') }
+    stub_request(:any, userdata_uri).with { |request| !request.headers.key?('X-aws-ec2-metadata-token') }
```

Comment:
stub the specific request type(`:get`) to reflect the use case of this scenarion